storage-proofs/porep/src/stacked/vanilla/create_label/multi.rs

```diff
@@ -439,14 +439,13 @@ pub fn create_labels_for_encoding<Tree: 'static + MerkleTreeTrait, T: AsRef<[u8]
 
     let core_group = Arc::new(checkout_core_group());
 
-    if let Some(group) = &*core_group {
+    // When `_cleanup_handle` is dropped, the previous binding of thread will be restored.
+    let _cleanup_handle = (*core_group).as_ref().map(|group| {
```

Comment:
you need this in the `decoding` one too

Reply:
Thanks, will add. It would be really nice if we could unify those paths, exactly for reasons like this. I'll look into it later.